docs/core/testing/unit-testing-mstest-migration-from-v1-to-v3.md

```diff
@@ -0,0 +1,290 @@
+---
+title: MSTest migration to v3
+description: Migration guide.
+author: engyebrahim
+ms.author: enjieid
+ms.date: 11/06/2024
+---
+
+# MSTest v3 migration guide
+
+## Overview
+
+This guide assists users in upgrading their MSTest projects from MSTest v1 to MSTest v3. MSTest v3 introduces significant new features, optimizations, and some breaking changes to improve test reliability, execution speed, and compatibility with modern .NET frameworks.
+
+---
+
+### Who Is Impacted?
+
+This guide is intended for projects currently using MSTest v1 via either:
+
+- **Assembly references**: Projects referencing `Microsoft.VisualStudio.QualityTools.UnitTestFramework.dll` directly in their project files.
+- **NuGet packages**: Projects using MSTest v1 through the `MSTest.TestFramework` NuGet package.
+
+If your project relies on MSTest for unit testing and includes the above references, it will benefit from the improvements in MSTest v3 and requires adjustments outlined in this guide.
+
+---
+
+### Why Migrate to MSTest v3?
+
+Even if you’re satisfied with your current MSTest setup, upgrading to MSTest v3 unlocks substantial advantages that improve both the quality and future-readiness of your tests. Here’s why making the switch now can be a valuable step forward:
+
+- **Immediate Performance Boosts**: MSTest v3 significantly reduces test execution time and optimizes resource usage. This is particularly beneficial in CI/CD pipelines, where faster tests can shorten deployment cycles and cut infrastructure costs.
```

Comment:
Is mstest v1 unsupported by us? Or at best only getting security fixes? Maybe that should be the first point.

Reply:
Good point @nohwnd, I think it's worth starting with the fact that's considered unsecured.